frontend/src/Leaderboard/RowTopTen.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  View, Image, Text,
+  StyleSheet,
+} from 'react-native';
+import { SubCardView } from '../common/components';
+import goldIcon from '../../assets/gold.png';
+import crownIcon from '../../assets/crown.png';
+import userAvatar from '../../assets/user-avatar.png';
+import Colors from '../common/colors';
+
+const styles = StyleSheet.create({
+  text: {
+    fontSize: 14,
+    color: Colors.whiteColor,
+  },
+  userAvatar: {
+    width: 28,
+    height: 28,
+    marginHorizontal: 8,
+  },
+  row: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    paddingVertical: 4,
+  },
+  icon: {
+    width: 20,
+    height: 20,
+    marginHorizontal: 4,
+  },
+  rowFlex: {
+    flexDirection: 'row',
+    alignItems: 'center',
+  },
+  userInfo: {
+    flexDirection: 'row',
+    alignItems: 'center',
+    marginRight: 12,
+  },
+});
+
+function RowTopTen({
+  rank, username, gold, crown,
+}) {
+  return (
+    <SubCardView style={styles.row}>
+      <View style={styles.rowFlex}>
+        <Text style={styles.text}>{rank}</Text>
+        <Image style={styles.userAvatar} source={userAvatar} />
+        <Text style={styles.text}>{username}</Text>
```

Comment:
if the username is too long, the amount of gold won't be seen. Please fix this. For this you'll need to change on the styling just a little bit, give the text a width, and because of this you can't use the text styling for all of the texts here. 
Handling too long text is really easy in React Native, it's only adding one attribute to the Text:) let me know if you need help